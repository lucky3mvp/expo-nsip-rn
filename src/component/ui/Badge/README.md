---
title: Badge
subTitle: 徽章
---


# Badge组件
### 组件描述
- 用于展示更新提示的红点或者数字,实现参考Antd-Mobile Badge组件


### 示例代码

```html
import Badge from 'Badge'
<Badge />
```

## Props(待做)
|属性名|属性功能|属性Value|默认Value|
|--|--|--|--|
|size|尺寸|||
|dot|是否只显示红点|||
|text|Badge内容：如数字|||
