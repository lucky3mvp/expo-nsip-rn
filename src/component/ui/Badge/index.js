/**
 * Badge组件: 用于显示有新消息的视觉元素，可以显示为红点或是数字
 * 目前只实现绘制红点的功能，位置需要通过额外style设置
 */
import React from 'react'
import { 
	View,
	StyleSheet
} from 'react-native'
import UIComponent from '../../common/UIComponent'

export default class Badge extends UIComponent {
	static defaultProps = {
	}

	static propTypes = {
	}

	constructor(props){
		super(props)
	}

	render(){
		let style = this.style
		let { children } = this.props;

		return (
			<View style={style.container}>
				{children}
			</View>
		)
	}
}

const dotSize = 3
Badge.baseStyle = {
	container: {
		width: 2 * dotSize,
		height: 2* dotSize,
		borderRadius: dotSize,
		backgroundColor: '#F05B48',
	}
}