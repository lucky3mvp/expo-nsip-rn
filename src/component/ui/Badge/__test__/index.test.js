import React from 'react'
import Renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Badge from '../index'

describe('Badge Tests', () => {
	it('renders correctly', () => {
		const tree = Renderer.create((
			<Badge />
		)).toJSON();
		
		expect(tree).toMatchSnapshot();
	});
})
