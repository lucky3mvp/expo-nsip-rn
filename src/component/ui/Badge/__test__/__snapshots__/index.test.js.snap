// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Tests renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F05B48",
      "borderRadius": 3,
      "height": 6,
      "width": 6,
    }
  }
/>
`;
